freeipa/src/main/java/com/sequenceiq/freeipa/service/config/RootCertRegisterService.java

```diff
@@ -68,6 +63,34 @@ public RootCert register(Stack stack) throws FreeIpaClientException {
         return registerInternal(stack);
     }
 
+    @Override
+    public void delete(Stack stack) {
+        try {
+            rootCertService.deleteByStack(stack);
+        } catch (NotFoundException e) {
+            LOGGER.info("Root cert not exists for environment {}", stack.getEnvironmentCrn());
+        }
+    }
+
+    private RootCert registerInternal(Stack stack) throws FreeIpaClientException {
+        MDCBuilder.buildMdcContext(stack);
+        String rootCertificate = getRootCertFromFreeIpa(stack);
+        Optional<RootCert> currentCert = rootCertService.findByStackId(stack.getId());
+        return currentCert.isPresent() ? updateRootCertEntityIfNecessary(currentCert.get(), rootCertificate) : createRootCertEntity(stack, rootCertificate);
+    }
+
+    private RootCert updateRootCertEntityIfNecessary(RootCert rootCert, String rootCertificate) {
+        String rootCertInPemFormat = convertToPemFormat(rootCertificate);
+        if (!rootCert.getCert().equals(rootCertInPemFormat)) {
```

Comment:
Potential NPE if `rootCert.getCert() == null`. As `rootCertInPemFormat` can itself never be `null`, I suggest swapping the invocation target and the method arg.

Reply:
you're somewhat right, it could be null in theory, but I don't think we are persisting any null there, but sure, switching is not a big deal and it would make it safer.

Reply:
Thanks! Done